www/js/app.js

```diff
@@ -55,10 +52,44 @@ var app = angular.module('starter',
                 url:'/settings',
                 templateUrl: 'templates/settings.html',
                 controller: 'Settings'
+            })
+            .state('profile', {
+                url:'/profile',
+                templateUrl: 'templates/profile.html',
+                controller: 'Profile'
             });
         $urlRouterProvider.otherwise('/');
     }) 
     
+    .directive('profileBarWidget', function($ionicGesture) {
+        return {
+            restrict: 'AE',
+            // declare the directive scope as private (and empty)
+            scope: {},
+            // add behaviour to our buttons and use a variable value
+            template:   '<div class="thumbnail">'+
+                            '<img src="img/ionic.png" style="width:60px;height:60px;"'+
+                                'alt="Not By Design" />'+
+                        '</div>'+
+                        '<div class="profile-name">'+       
+                            '<h3> {{ profile.name }} </h3>'+
+                        '</div>'+
+                        '<div class="edit-button">'+
+                            '<button class="button button-positive button-small"'+
+                               'onclick="window.location=\'#/profile\'">Edit</button>'+
+                        '</div>',
+            // we just declare what we need in the above template
+            controller: function(userInfoService, $scope) {
+                userInfoService.getProfileInfo().then(function(userInfo) {
```

Comment:
So make sure that you have added the new .js file to the index.html so it gets loaded in and that you are declaring the new controller correctly (although if you already moved the directive out into a new file, then you can just declare it inline as you already have).